### ISSUE TITLE
[Identity] Support EnvironmentCredential from Azure Identity

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-cli#14544]
> Originally reported in #9282, #10241, #10903, https://github.com/Azure/login/issues/27. An attempt was made to implement it in https://github.com/Azure/azure-cli/pull/12867 but that PR got delayed due to the Azure Identity integration.

When using `az login` with `--password`, it is possible for malicious program to retrieve the command line parameter, including the password. For example, this can be done using the [`ps`](https://man7.org/linux/man-pages/man1/ps.1.html) command.

``` 
$ export SP_PASSWORD="something"
$ az login -p $SP_PASSWORD &
[1] 21
$ ps -f
UID        PID  PPID  C STIME TTY          TIME CMD
user2        9     8  0 13:04 tty1     00:00:00 -bash
user2       21     9 21 13:04 tty1     00:00:01 /usr/bin/python3 -m azure.cli login -p something
user2       25     9  0 13:04 tty1     00:00:00 ps -f
```
 
The password `something` is exposed in the output.

A better choice is to utilize [`EnvironmentCredential`](https://azuresdkdocs.blob.core.windows.net/web/python/azure-identity/1.2.0/azure.identity.html#azure.identity.EnvironmentCredential) from Azure Identity so that this information is hidden and the login process can be more secure.

**Comment by jiasli** [comment]
A workaround is to use a password file through Azure CLI's [`@` convention](https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#generic-update-arguments):
 
```
az login --service-principal -u <app-url> -p @password.txt --tenant <tenant>
```

**Comment by nadesu** [comment]
@jiasli password file based usage of CLI is also risk prone as the password is now available in file and there is risk of missing the file during cleanup. After consulting ADO security team, its advised to use Environment credential approach (credential configured by environment variables). 
Let us know the ETA for availability of this feature.

**Comment by aksm-ms** [comment]
@jiasli Is Environment credential support added to az cli login? cc: @nadesu

**Comment by jiasli** [comment]
We will release `EnvironmentCredential` in next beta version.

**Comment by ColonelPanicks** [comment]
Any update on when the next beta version will be released? Or when the environment variables will become available?

According to the [CLI Beta Release Notes](https://docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?tabs=azure-cli-beta) there hasn't been a beta version since June 2020.

I found [this page](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/docs-ref-conceptual/install-azure-cli-beta.md#install-azure-cli-beta) which indicates how to install the beta version on Linux but it doesn't look to me like the environment variable authentication is working in it (after exporting the variables mentioned [here](https://github.com/Azure/azure-cli/pull/12867).

Slightly unrelated note but the beta install doc linked above doesn't seem to appear on the website for the docs? It looks like beta release installs are only [described for Windows](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?tabs=azure-cli#azure-cli-beta-version).

**Comment by AmrutaKawade** [comment]
@jiasli is this releases?

**Comment by ColonelPanicks** [comment]
Any update on this? It would be really useful to be able to export environment variables for authentication.

It looks like _something_ akin to this [was implemented in 2.53.1](https://github.com/Azure/azure-cli/commit/1d973cceb38980181eeaa45934497d2148a3e7b2) however that seems to only apply to rdbms/mysql authentication? Despite #14960 supposedly addressing authentication in general with the identity service.

@jiasli can you elaborate on the state of this request? Seems misleading that this was closed as completed in Oct 2020 but hasn't actually been completed. 

My attempts at using it don't work as expected:
```bash
[root@aztest1 ~]# export AZURE_TENANT_ID=REDACTED
[root@aztest1 ~]# export AZURE_CLIENT_ID=REDACTED
[root@aztest1 ~]# export AZURE_CLIENT_SECRET='REDACTED'
[root@aztest1 ~]# az account show
Please run 'az login' to setup account.

[root@aztest1 ~]# export TENANT_ID=REDACTED
[root@aztest1 ~]# export CLIENT_ID=REDACTED
[root@aztest1 ~]# export CLIENT_SECRET='REDACTED'
[root@aztest1 ~]# az account show
Please run 'az login' to setup account.
```

Am I doing something wrong here?
